balances/src/impl_fungibles.rs

```diff
@@ -0,0 +1,180 @@
+use super::*;
+use encointer_primitives::{common::PalletString, fixed::transcendental::powi};
+use frame_support::{
+	inherent::Vec,
+	traits::{
+		fungibles::InspectMetadata,
+		tokens::{DepositConsequence, WithdrawConsequence},
+	},
+};
+use sp_runtime::{traits::Zero, DispatchError};
+
+impl<T: Config> fungibles::InspectMetadata<T::AccountId> for Pallet<T> {
+	fn name(asset: &Self::AssetId) -> Vec<u8> {
+		PalletString::from("Encointer").into()
+	}
+
+	fn symbol(asset: &Self::AssetId) -> Vec<u8> {
+		PalletString::from("ETR").into()
+	}
+
+	fn decimals(asset: &Self::AssetId) -> u8 {
+		// Our BalanceType is I64F64, so the smallest possible number is 2^-64 = 5.42101086242752217003726400434970855712890625 × 10^-20
+		//  and an upper bound is 2^63 + 1 = 9.223372036854775809 × 10^18
+		// so we chose 18 decimals and lose some precision but can prevent overflows that way.
+		18u8
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	pub fn balance_type_to_fungible_balance(
+		asset: <Pallet<T> as fungibles::Inspect<T::AccountId>>::AssetId,
+		balance: BalanceType,
+	) -> <Pallet<T> as fungibles::Inspect<T::AccountId>>::Balance {
+		assert!(balance >= BalanceType::from_num(0));
+		let decimals = Self::decimals(&asset);
+
+		let bits = balance.to_bits();
+		let mut result: u128 = 0;
+
+		result = result + (bits >> 64) as u128 * 10u128.pow(decimals as u32);
+
+		result = result +
+			(BalanceType::from_bits((bits as i64) as i128) *
+				BalanceType::from_num(10u128.pow(decimals as u32)))
+			.to_num::<u128>();
+		result
+	}
+
+	pub fn fungible_balance_to_balance_type(
+		asset: <Pallet<T> as fungibles::Inspect<T::AccountId>>::AssetId,
+		fungible_balance: <Pallet<T> as fungibles::Inspect<T::AccountId>>::Balance,
+	) -> BalanceType {
+		let decimals = Self::decimals(&asset);
+		let mut result: BalanceType = BalanceType::from_num(0);
+
+		result = result +
+			BalanceType::from_num(
+				((fungible_balance << 64) >> 64) as f64 / (10i128.pow(decimals as u32) as f64),
+			);
```

Comment:
`((fungible_balance << 64) >> 64)` I guess this is to truncate, we might find some higher level function for this. :)

Reply:
This is actually the same as `fungible_balance as u64`